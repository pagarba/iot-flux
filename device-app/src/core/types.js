export default {
  LOGIN_REQUEST: 'LOGIN_REQUEST',
  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
  LOGIN_FAILURE: 'LOGIN_FAILURE',
  LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
  LOGOUT_FAILURE: 'LOGOUT_FAILURE',
  ACCOUNT_FETCH_SUCCESS: 'ACCOUNT_FETCH_SUCCESS',

  CREATE_USER_REQUEST: 'CREATE_USER_REQUEST',
  CREATE_USER_SUCCESS: 'CREATE_USER_SUCCESS',
  CREATE_USER_FAILURE: 'CREATE_USER_FAILURE',

  CREATE_DEVICE_REQUEST: 'CREATE_DEVICE_REQUEST',
  CREATE_DEVICE_SUCCESS: 'CREATE_DEVICE_SUCCESS',
  CREATE_DEVICE_FAILURE: 'CREATE_DEVICE_FAILURE',

  EDIT_DEVICE_REQUEST: 'EDIT_DEVICE_REQUEST',
  EDIT_DEVICE_SUCCESS: 'EDIT_DEVICE_SUCCESS',
  EDIT_DEVICE_FAILURE: 'EDIT_DEVICE_FAILURE',

  GET_DEVICES_REQUEST: 'GET_DEVICES_REQUEST',
  GET_DEVICES_SUCCESS: 'GET_DEVICES_SUCCESS',
  GET_DEVICES_FAILURE: 'GET_DEVICES_FAILURE',

  GET_DEVICE_REQUEST: 'GET_DEVICE_REQUEST',
  GET_DEVICE_SUCCESS: 'GET_DEVICE_SUCCESS',
  GET_DEVICE_FAILURE: 'GET_DEVICE_FAILURE',

  DELETE_DEVICE_REQUEST: 'DELETE_DEVICE_REQUEST',
  DELETE_DEVICE_SUCCESS: 'DELETE_DEVICE_SUCCESS',
  DELETE_DEVICE_FAILURE: 'DELETE_DEVICE_FAILURE',

  CREATE_CHANNEL_REQUEST: 'CREATE_CHANNEL_REQUEST',
  CREATE_CHANNEL_SUCCESS: 'CREATE_CHANNEL_SUCCESS',
  CREATE_CHANNEL_FAILURE: 'CREATE_CHANNEL_FAILURE',

  EDIT_CHANNEL_REQUEST: 'EDIT_CHANNEL_REQUEST',
  EDIT_CHANNEL_SUCCESS: 'EDIT_CHANNEL_SUCCESS',
  EDIT_CHANNEL_FAILURE: 'EDIT_CHANNEL_FAILURE',

  GET_CHANNELS_REQUEST: 'GET_CHANNELS_REQUEST',
  GET_CHANNELS_SUCCESS: 'GET_CHANNELS_SUCCESS',
  GET_CHANNELS_FAILURE: 'GET_CHANNELS_FAILURE',

  GET_CHANNEL_REQUEST: 'GET_CHANNEL_REQUEST',
  GET_CHANNEL_SUCCESS: 'GET_CHANNEL_SUCCESS',
  GET_CHANNEL_FAILURE: 'GET_CHANNEL_FAILURE',

  DELETE_CHANNEL_REQUEST: 'DELETE_CHANNEL_REQUEST',
  DELETE_CHANNEL_SUCCESS: 'DELETE_CHANNEL_SUCCESS',
  DELETE_CHANNEL_FAILURE: 'DELETE_CHANNEL_FAILURE',

  ADD_DEVICE_TO_CHANNEL_REQUEST: 'ADD_DEVICE_TO_CHANNEL_REQUEST',
  ADD_DEVICE_TO_CHANNEL_SUCCESS: 'ADD_DEVICE_TO_CHANNEL_SUCCESS',
  ADD_DEVICE_TO_CHANNEL_FAILURE: 'ADD_DEVICE_TO_CHANNEL_FAILURE',

  DELETE_DEVICE_FROM_CHANNEL_REQUEST: 'DELETE_DEVICE_FROM_CHANNEL_REQUEST',
  DELETE_DEVICE_FROM_CHANNEL_SUCCESS: 'DELETE_DEVICE_FROM_CHANNEL_SUCCESS',
  DELETE_DEVICE_FROM_CHANNEL_FAILURE: 'DELETE_DEVICE_FROM_CHANNEL_FAILURE',

  GET_CHANNEL_DEVICES_REQUEST: 'GET_CHANNEL_DEVICES_REQUEST',
  GET_CHANNEL_DEVICES_SUCCESS: 'GET_CHANNEL_DEVICES_SUCCESS',
  GET_CHANNEL_DEVICES_FAILURE: 'GET_CHANNEL_DEVICES_FAILURE',

  CREATE_EVENT_REQUEST: 'CREATE_EVENT_REQUEST',
  CREATE_EVENT_SUCCESS: 'CREATE_EVENT_SUCCESS',
  CREATE_EVENT_FAILURE: 'CREATE_EVENT_FAILURE',

  EDIT_EVENT_REQUEST: 'EDIT_EVENT_REQUEST',
  EDIT_EVENT_SUCCESS: 'EDIT_EVENT_SUCCESS',
  EDIT_EVENT_FAILURE: 'EDIT_EVENT_FAILURE',

  GET_EVENTS_REQUEST: 'GET_EVENTS_REQUEST',
  GET_EVENTS_SUCCESS: 'GET_EVENTS_SUCCESS',
  GET_EVENTS_FAILURE: 'GET_EVENTS_FAILURE',

  GET_EVENT_REQUEST: 'GET_EVENT_REQUEST',
  GET_EVENT_SUCCESS: 'GET_EVENT_SUCCESS',
  GET_EVENT_FAILURE: 'GET_EVENT_FAILURE',

  DELETE_EVENT_REQUEST: 'DELETE_EVENT_REQUEST',
  DELETE_EVENT_SUCCESS: 'DELETE_EVENT_SUCCESS',
  DELETE_EVENT_FAILURE: 'DELETE_EVENT_FAILURE',
}