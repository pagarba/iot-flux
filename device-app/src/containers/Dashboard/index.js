import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import { Grid, Typography } from '@material-ui/core';

import { Link } from 'react-router-dom';

import ChannelsIcon from '@material-ui/icons/GridOnOutlined';
import AddDeviceIcon from '@material-ui/icons/AddToQueue';
import ViewEventsIcon from '@material-ui/icons/VisibilityOutlined';
import TutorialIcon from '@material-ui/icons/SchoolOutlined';
import ShoppingCartIcon from '@material-ui/icons/ShoppingCartOutlined';
import PanToolIcon from '@material-ui/icons/PanTool';

const styles = () => ({
  dashboard: {
    textAlign: 'center',
    marginTop: '50px'
  },
  icon: {
    fontSize: '80px',
  },
  link: {
    textDecoration: 'none',
    color: 'rgba(0, 0, 0, 0.87)'
  },
  text: {
    fontSize: '16px',
    textTransform: 'uppercase',
  },
  root: {
    flexGrow: 1,
  },
});

export class Dashboard extends Component {
  constructor() {
    super();
  }

  render() {
    const { classes } = this.props;

    return (
      <div className={classes.dashboard}>
        <Grid container spacing={24}>
          <Grid item xs={6} sm={4} md={2}>
            <Link className={classes.link} to="/marketplace?mode=create">
              <Grid item xs container direction="column" spacing={16}>
                <Grid item>
                  <ShoppingCartIcon className={classes.icon}/>
                </Grid>
                <Grid item>
                  <Typography className={classes.text} color="secondary" gutterBottom>Marketplace</Typography>
                </Grid>
              </Grid>
            </Link>
          </Grid>
          <Grid item xs={6} sm={4} md={2}>
            <Link className={classes.link} to="/build-kits?mode=create">
              <Grid item xs container direction="column" spacing={16}>
                <Grid item>
                  <PanToolIcon className={classes.icon}/>
                </Grid>
                <Grid item>
                  <Typography className={classes.text} color="secondary" gutterBottom>Build Kits</Typography>
                </Grid>
              </Grid>
            </Link>
          </Grid>
          <Grid item xs={6} sm={4} md={2}>
            <Link className={classes.link} to="/channels?mode=create">
              <Grid item xs container direction="column" spacing={16}>
                <Grid item>
                  <ChannelsIcon className={classes.icon}/>
                </Grid>
                <Grid item>
                  <Typography className={classes.text} color="secondary" gutterBottom>Add Channel</Typography>
                </Grid>
              </Grid>
            </Link>
          </Grid>
          <Grid item xs={6} sm={4} md={2}>
            <Link className={classes.link} to="/devices?mode=create">
              <Grid item xs container direction="column" spacing={16}>
                <Grid item>
                  <AddDeviceIcon className={classes.icon}/>
                </Grid>
                <Grid item>
                  <Typography className={classes.text} color="secondary" gutterBottom>Add Device</Typography>
                </Grid>
              </Grid>
            </Link>
          </Grid>
          <Grid item xs={6} sm={4} md={2}>
            <Link className={classes.link} to="/events?mode=create">
              <Grid item xs container direction="column" spacing={16}>
                <Grid item>
                  <ViewEventsIcon className={classes.icon}/>
                </Grid>
                <Grid item>
                  <Typography className={classes.text} color="secondary" gutterBottom>View Events</Typography>
                </Grid>
              </Grid>
            </Link>
          </Grid>
          <Grid item xs={6} sm={4} md={2}>
            <Link className={classes.link} to="/tutorials?mode=create">
              <Grid item xs container direction="column" spacing={16}>
                <Grid item>
                  <TutorialIcon className={classes.icon}/>
                </Grid>
                <Grid item>
                  <Typography className={classes.text} color="secondary" gutterBottom>Tutorial</Typography>
                </Grid>
              </Grid>
            </Link>
          </Grid>
        </Grid>
      </div>
    );
  }
}

Dashboard.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Dashboard);
