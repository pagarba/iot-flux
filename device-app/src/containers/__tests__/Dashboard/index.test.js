import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { Dashboard } from '../../Dashboard';

describe('<Dashboard />', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it('should render components', () => {
    const wrapper = shallow(
      <Dashboard
        classes={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});