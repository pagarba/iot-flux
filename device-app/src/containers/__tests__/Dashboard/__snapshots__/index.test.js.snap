// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should render components 1`] = `
<div>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      md={2}
      sm={4}
      xs={6}
    >
      <Link
        replace={false}
        to="/marketplace?mode=create"
      >
        <WithStyles(Grid)
          container={true}
          direction="column"
          item={true}
          spacing={16}
          xs={true}
        >
          <WithStyles(Grid)
            item={true}
          >
            <pure(ShoppingCartOutlinedIcon) />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            <WithStyles(Typography)
              color="secondary"
              gutterBottom={true}
            >
              Marketplace
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </Link>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={2}
      sm={4}
      xs={6}
    >
      <Link
        replace={false}
        to="/build-kits?mode=create"
      >
        <WithStyles(Grid)
          container={true}
          direction="column"
          item={true}
          spacing={16}
          xs={true}
        >
          <WithStyles(Grid)
            item={true}
          >
            <pure(PanToolIcon) />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            <WithStyles(Typography)
              color="secondary"
              gutterBottom={true}
            >
              Build Kits
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </Link>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={2}
      sm={4}
      xs={6}
    >
      <Link
        replace={false}
        to="/channels?mode=create"
      >
        <WithStyles(Grid)
          container={true}
          direction="column"
          item={true}
          spacing={16}
          xs={true}
        >
          <WithStyles(Grid)
            item={true}
          >
            <pure(GridOnOutlinedIcon) />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            <WithStyles(Typography)
              color="secondary"
              gutterBottom={true}
            >
              Add Channel
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </Link>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={2}
      sm={4}
      xs={6}
    >
      <Link
        replace={false}
        to="/devices?mode=create"
      >
        <WithStyles(Grid)
          container={true}
          direction="column"
          item={true}
          spacing={16}
          xs={true}
        >
          <WithStyles(Grid)
            item={true}
          >
            <pure(AddToQueueIcon) />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            <WithStyles(Typography)
              color="secondary"
              gutterBottom={true}
            >
              Add Device
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </Link>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={2}
      sm={4}
      xs={6}
    >
      <Link
        replace={false}
        to="/events?mode=create"
      >
        <WithStyles(Grid)
          container={true}
          direction="column"
          item={true}
          spacing={16}
          xs={true}
        >
          <WithStyles(Grid)
            item={true}
          >
            <pure(VisibilityOutlinedIcon) />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            <WithStyles(Typography)
              color="secondary"
              gutterBottom={true}
            >
              View Events
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </Link>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={2}
      sm={4}
      xs={6}
    >
      <Link
        replace={false}
        to="/tutorials?mode=create"
      >
        <WithStyles(Grid)
          container={true}
          direction="column"
          item={true}
          spacing={16}
          xs={true}
        >
          <WithStyles(Grid)
            item={true}
          >
            <pure(SchoolOutlinedIcon) />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            <WithStyles(Typography)
              color="secondary"
              gutterBottom={true}
            >
              Tutorial
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </Link>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
